### PR TITLE
Type BuildAndInstallRequest.

### PR DESCRIPTION
**Pull request by jsirois**
Work towards #899.

### CODE REVIEW

**Comment by jsirois** [comment]
Depends on #1167 (2f2fadd) and #1168 (2229d24). Please just review 3ebfda2.

File: pex/resolver.py [hunk 1]
```diff
@@ -347,32 +376,46 @@ def from_request(cls, build_request, dist_root):
...
     def finalize_build(self):
-        self.atomic_dir.finalize()
+        # type: () -> Iterator[InstallRequest]
+        self._atomic_dir.finalize()
         for wheel in os.listdir(self.dist_dir):
             yield InstallRequest.create(self.request.target, os.path.join(self.dist_dir, wheel))
 
 
-class InstallRequest(namedtuple("InstallRequest", ["target", "wheel_path", "fingerprint"])):
+class InstallRequest(object):
```
Comment (Eric-Arellano) [inline pex/resolver.py:410]: Does this need `__eq__` and `__hash__`? May be worth doing `__repr__` too.

Comment (Eric-Arellano) [reply pex/resolver.py:410]: Ditto on the other classes in this file that don't use namedtuple anymore.

**Review by Eric-Arellano — APPROVED** [review]
Excellent! Thanks!

File: pex/resolver.py (hunk 1 above)
Comment (jsirois) [reply pex/resolver.py:410]: They do not need `__eq__` / `__hash__` but `{Build,Install}Request` could both stand a `__repr__` for the verbose logs.

**Pull request merged**